test/runtests.jl

```diff
@@ -1,9 +1,22 @@
 using TulipaProfileFitting
 using Test
 
-@testset "TulipaProfileFitting.jl" begin
+function test_fit_profile()
+
+    # Test case 1: Using root finding method
     profile_values = [0.2, 0.5, 0.9, 1.0, 0.4, 0.1, 0.0]
     target_mean = 0.5
-    coefficient = fit_profile(profile_values, target_mean)
-    @test coefficient ≈ 0.7491921789832345 atol = 0.01
+    expected_coefficient = 0.7491921797062476
+    @test fit_profile(profile_values, target_mean) ≈ expected_coefficient atol = 1e-3
+
+    # Test case 2: Using optimization method
+    target_mean = 0.4
+    expected_coefficient = 1.2550552057968192
+    @test fit_profile(profile_values, target_mean) ≈ expected_coefficient atol = 1e-3
+end
```

Comment:
It makes sense to move this inside the `@testset`. The scope will also be limited.